py/src/braintrust/logger.py

```diff
@@ -791,13 +794,17 @@ def decorator(span_args, span_kwargs, f):
 
         @wraps(f)
         def wrapper_sync(*f_args, **f_kwargs):
-            with start_span(*span_args, **span_kwargs):
-                return f(*f_args, **f_kwargs)
+            with start_span(*span_args, **span_kwargs) as span:
```

Comment:
Can we put these changes in a separate PR? I am worried about having substantially different behavior between the python and javascript SDKs, and I don't want to pollute this PR with that conversation.

Reply:
Yep

Reply:
https://github.com/braintrustdata/braintrust-sdk/pull/33